penrose-web/src/engine/EngineUtils.ts

```diff
@@ -175,25 +214,28 @@ function mapPalette<T, S>(f: (arg: T) => S, v: IPaletteV<T>): IPaletteV<S> {
 
 // Expects `f` to be a function between numeric types (e.g. number -> VarAD, VarAD -> number, AD var -> VarAD ...)
 // Coerces any non-numeric types
-export function mapValueNumeric<T, S>(f: (arg: T) => S, v: Value<T>): Value<S> {
-  const nonnumericValueTypes = [
-    "IntV",
-    "BoolV",
-    "StrV",
-    "ColorV",
-    "PaletteV",
-    "FileV",
-    "StyleV",
-  ];
+export function mapValueNumeric<T, S>(
+  f: (arg: T) => S,
+  v: Value<T>
+): Value<S> {
+  const nonnumericValueTypes = ["BoolV", "StrV", "ColorV", "PaletteV", "FileV", "StyleV"];
```

Comment:
Maybe we can make this a type? I bet this will be useful for the evaluator/compiler when it's trying to find (non)numeric types.

Reply:
Sure, I can do that. But how would I use that type as in L239 in `EngineUtils` (when I need to check if a tag is a member of that set of types)? Not sure how to do that in TypeScript.

Reply:
How about [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)?

Reply:
Let's hold off on doing this until there's a second use case for `nonnumericValueTypes`.